smithy-model/src/main/java/software/amazon/smithy/model/shapes/SmithyIdlModelSerializer.java

```diff
@@ -532,9 +532,13 @@ private void serializeTraits(Map<ShapeId, Trait> traits, TraitFeature... traitFe
 
         private void serializeDocumentation(String documentation) {
             // The documentation trait has a special syntax, which we always want to use.
-            codeWriter.setNewlinePrefix("/// ")
+            codeWriter
+                    .pushState()
+                    // See https://github.com/smithy-lang/smithy/issues/2115
+                    .trimTrailingSpaces(false)
+                    .setNewlinePrefix("/// ")
                     .write(documentation.replace("$", "$$"))
```

Comment:
**question:** what's the motivation for this replace btw? I didn't think `$` had any special meaning in doc comments...

Reply:
The write thing this goes through treats them specially as placeholders, so we have to escape them. The writer later turns it back into "$".